src/main/kotlin/com/launchdarkly/intellij/toolwindow/LinkPanel.kt

```diff
@@ -0,0 +1,134 @@
+
+    private fun initTree(model: AsyncTreeModel): Tree {
+        tree = Tree(model)
+        tree.isRootVisible = false
+        FlagTreeSearch(tree)
+        TreeUtil.installActions(tree)
+        tree.selectionModel.selectionMode = TreeSelectionModel.SINGLE_TREE_SELECTION
+
+        return tree
+    }
+
+    fun start(): Tree {
+        var reviewTreeBuilder = AsyncTreeModel(treeModel, this)
+        tree = initTree(reviewTreeBuilder)
+
+        val componentsSplitter = OnePixelSplitter(SPLITTER_PROPERTY, 0.33f)
+        componentsSplitter.setHonorComponentsMinimumSize(true)
+        componentsSplitter.setHonorComponentsMinimumSize(true)
+        componentsSplitter.firstComponent = JPanel(CardLayout()).apply {
+            add(ScrollPaneFactory.createScrollPane(tree, SideBorder.NONE), "Tree")
+        }
+        setContent(componentsSplitter)
+        return tree
+    }
+
+    private val listener = object : MouseAdapter() {
+        override fun mouseClicked(e: MouseEvent) {
+            if (e.button != BUTTON1) {
+                return
+            }
+            val tree = e.component as Tree
+            val node = tree.lastSelectedPathComponent as? DefaultMutableTreeNode ?: return
+            val userNode = node.userObject as LinkNodeBase
+            BrowserLauncher.instance.open(userNode.url)
+            tree.clearSelection()
+        }
+    }
+
+    fun actions(tree: Tree) {
+        tree.addMouseListener(listener)
+    }
+
+    fun updateNodeInfo() {
+        root = LinkNodeRoot(settings)
+        treeStructure = createTreeStructure()
+        treeModel = StructureTreeModel(treeStructure, this)
+        var reviewTreeBuilder = AsyncTreeModel(treeModel, this)
+        tree.model = reviewTreeBuilder
+    }
+
+    init {
+        if (settings.isConfigured()) {
+            tree = start()
+            actions(tree)
+        }
+        var start = false
+        if (!this::tree.isInitialized) {
+            start = true
+        }
+
+        try {
+            myProject.messageBus.connect().subscribe(
+                messageBusService.flagsUpdatedTopic,
+                object : FlagNotifier {
+                    override fun notify(isConfigured: Boolean, flag: String, rebuild: Boolean) {
+                    }
+
+                    override fun reinit() {
+                        if (start) {
+                            tree = start()
+                            actions(tree)
+                        }
+                        UIUtil.invokeLaterIfNeeded {
+                            updateNodeInfo()
+                        }
+                    }
+                }
+            )
+        } catch (err: Error) {
+            System.err.println("Exception when updating LaunchDarkly LinkPanel Toolwindow")
+            err.printStackTrace()
+            val notifier = GeneralNotifier()
+            notifier.notify(
+                myProject,
+                "Error updating LaunchDarkly Toolwindow $err"
```

Comment:
This is another opportunity to DRY because I see the same error catching logic in the [FlagPanel](https://github.com/launchdarkly/ld-intellij/blob/df155a9d08e39833f51791383dd714d21ec332c6/src/main/kotlin/com/launchdarkly/intellij/toolwindow/FlagPanel.kt#L261). We can write an error notifier function which can be shared across the project.